Feature clean target

mwieder opened the pull request:
Got tired of trying to get the syntax right. We really should have a "make clean" target anyway.

montegoulding commented:
@mwieder it appears you have inadvertently committed a heap of extra files from the prebuilt directory. They may just be legacy files from when it was a submodule... Not sure what's going on there but you may want to:

```
git checkout develop
rm -rf prebuilt
git reset HEAD --hard
```

Also given all the merge commits I suspect you have your `develop` branch tracking your fork `origin/develop` instead of ours `upstream/develop`. Unless you have set it (If unsure `git remote -v`) up you won't have an `upstream` remote yet so:

```
git remote add upstream https://github.com/livecode/livecode.git
git fetch --all
git checkout develop
git reset --hard upstream/develop
git branch -u upstream/develop
```

Once you have done that you can just do the following to update `develop`:

```
git checkout develop
git pull
```

As far as the patch goes once you have done the above it might be better to branch off the clean `develop` branch then:

```
git checkout -b feature_clean_target_b
git checkout 0927b9c Makefile
```
I am probably not the best person to review it but I strongly suspect you need to modify it so it does something like:

```
clean: clean-$(guess_platform)
```

Then add a `clean-<platform>` rule for each platform. It would also be good if `clean` also deleted the `prebuilt/fetched`,  `prebuilt/include` and `prebuilt/lib` for all platforms.

mwieder committed:
provides a clean target to remove errant binaries

mwieder commented:
Fixed the upstream setting done in a pre-caffeinated state. I changed the clean actions to include the legacy prebuilt directories, but I have concerns about doing a clean-$(guess_platform). My understanding is that would limit the cleanup to just the native platform. If I've got a 64-bit native platform and I'm compiling for 32-bit or for Android or something else, it wouldn't get scrubbed that way.

runrevmark commented on Makefile:
```diff
@@ -57,6 +57,15 @@ include Makefile.common
 check: check-$(guess_platform)
 
+# [[ MDW-2017-05-09 ]] feature_clean_target
+clean:
```
I suggest making this clean-linux for now - as it is specific to the linux builds.

mwieder committed:
changed target to clean-linux

runrevmark commented:
@livecode-vulcan review ok 445f6ce

livecodepanos commented:
@livecode-vulcan review ok 445f6c8

runrevmark commented:
@livecodepanos : Hehe - clearly I cannot read numbers today ;)

livecode-vulcan commented:
💙 review by @livecodepanos ok 445f6c8bc4ddd9c32acef20afb02b5896a3b1f5b

livecode-vulcan commented:
😎 test success 445f6c8bc4ddd9c32acef20afb02b5896a3b1f5b

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/243)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/252)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/232)
* try-community-universal-ios-iphoneos10.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos10.3/builds/175)
* try-community-universal-ios-iphonesimulator10.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator10.3/builds/175)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/241)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/246)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/245)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/250)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/245)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/250)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/269)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/240)
